bin/webpack-configs.js

```diff
@@ -262,14 +263,19 @@ const getMainConfig = ( options = {} ) => {
 					{
 						from: './assets/js/blocks/**/block.json',
 						to( { absoluteFilename } ) {
-							const blockName = absoluteFilename
+							/**
+							 * Getting the block name from the JSON metadata is less error prone
+							 * than extracting it from the file path.
+							 */
+							const JSONFile = fs.readFileSync(
+								path.resolve( __dirname, absoluteFilename )
+							);
+							const metadata = JSON.parse( JSONFile.toString() );
+							const blockName = metadata.name
 								.split( '/' )
-								.at( -2 );
+								.at( 1 );
 							return `./${ blockName }/block.json`;
 						},
```

Comment:
I assume we have little control into where this goes? currently, it pollutes our `build` folder with so many top level folders.
<img width="352" alt="image" src="https://user-images.githubusercontent.com/6165348/181820832-0e1284a0-a250-417d-818e-83f420cd9de2.png">


We actually have two folders there `checkout-blocks` and `cart-blocks`. Do you think we can output those block.json files there, and how much effort would it require? For block.json, we can change that in the future so this is not a blocker.